File: app/src/main/java/com/imangazalievm/bubbble/domain/interactors/UserDetailsInteractor.java
package com.imangazalievm.bubbble.domain.interactors;


import com.imangazalievm.bubbble.domain.models.User;
import com.imangazalievm.bubbble.domain.repository.UsersRepository;

import javax.inject.Inject;

import io.reactivex.Single;

public class UserDetailsInteractor {

    private UsersRepository usersRepository;

    @Inject
    public UserDetailsInteractor(UsersRepository usersRepository) {
        this.usersRepository = usersRepository;
    }


    public Single<User> getUser(long userId) {
        return usersRepository.getUser(userId);
    }

}

File: app-mvp/app/src/main/java/com/bubbble/domain/userprofile/UserDetailsInteractor.kt
package com.bubbble.domain.userprofile

import com.bubbble.data.users.UsersRepository
import com.bubbble.core.models.user.User
import javax.inject.Inject

class UserDetailsInteractor @Inject constructor(
    private val usersRepository: UsersRepository
) {

    suspend fun getUser(userId: Long): User {
        return usersRepository.getUser(userId)
    }

}